src/commands/mod.rs

```diff
@@ -261,7 +261,7 @@ impl DaemonControl {
         addr_info: &Option<AddrInfo>,
     ) {
         if let Some(AddrInfo { index, is_change }) = addr_info {
-            if *is_change && db_conn.change_index() < *index {
+            if *is_change && db_conn.change_index() <= *index {
```

Comment:
Should also fix the check against receive index right below?

Reply:
Ah yes, I'll take a look.

Reply:
I've now made this fix for the receive index and updated the functional tests.

I didn't test for this in the rust tests as `derivation_index_by_address` returns `None` in the dummy DB and so we don't enter the affected function in relation to receive addresses (we do enter the affected function for change addresses that have been generated within `create_spend`).